services/repository/src/main/java/org/sagebionetworks/repo/web/AccessInterceptor.java

```diff
@@ -51,9 +51,19 @@
 	private OIDCTokenHelper oidcTokenHelper;
 	
 	String getOAuthClientId(HttpServletRequest request) {
+		/*
+		 * There are two different places the client ID might be:
+		 *  - in the access token, if the OAuth client is acting on behalf of a user
+		 *  - in the basic auth credentials, if using an auth code/refresh token
+		 */
 		String accessToken = HttpAuthUtil.getBearerTokenFromStandardAuthorizationHeader(request);
-		if (accessToken==null) return null;
-		return oidcTokenHelper.parseJWT(accessToken).getBody().getAudience();
+		if (accessToken != null) {
+			return oidcTokenHelper.parseJWT(accessToken).getBody().getAudience();
+		} else if (HttpAuthUtil.usesBasicAuthentication(request)) {

```

Comment:
A small warning about this approach: this might not be enough since we use basic authentication in other filters that are not related to oauth and the "username" extracted might not actually be an OAuth client id (e.g. might be the user for docker registry or cloud mail in, a service call etc.). You might want to check for the special OAUTH_VERIFIED_CLIENT_ID_HEADER injected in the request instead.

Reply:
"+1". 

The two choices I see for changing this are (1) make sure it's the oauth client credentials rather than some other client credentials or (2) add a separate field, BASIC_AUTH_USER_NAME, which can be an oauth client or a non-oauth client.

Reply:
Thanks for the heads up, @marcomarasca ! It sounds like the right approach would be to use the injected header to conditionally insert the username into the OAuthClientId value, and as @brucehoff said, to log the basic auth username for all requests that include it.